07a_system_activity_plots.R

```diff
@@ -189,75 +190,107 @@ renderSystemPlot <- function(id) {
-      y = yend,
-      yend = yend
-    ),
-    linewidth = .3,
-    colour = "gray25",
-    linetype = "dashed",
-    show.legend = FALSE,
-    inherit.aes = FALSE
-  ) +
-  ggrepel::geom_text_repel(# the labels
-    aes(
-      x = group.id,
-      label = paste0(scales::comma(abs(values))),
-      y = rowSums(cbind(ystart, values / 2)),
-      segment.colour = "gray33"
-    ),
-    nudge_x = -.5,
-    arrow = arrow(type = "open", length = unit(.1, "inches")),
-    colour = "#4e4d47",
-    # alpha = .7,
-    size = 5,
-    inherit.aes = FALSE
-  ) +
-  scale_fill_manual(values = colors) + # color palette
-  scale_y_continuous(expand = c(0,0)) + # distance between bars and x axis line
-  scale_x_continuous(labels = str_wrap(df$Time %>% unique(), width = 10), # x axis labels
-                   breaks = df$group.id %>% unique()) +
-  theme_void() + # totally clear all theme elements
-  theme(# add back in what theme elements we want
-    text = element_text(size = 16, colour = "#4e4d47"),
-    axis.text.x = element_text(size = 16),
-    axis.ticks.x = element_line(),
-    axis.line.x = element_line(colour = "#4e4d47", linewidth = 0.5),
-    axis.ticks.length.x = unit(.15, "cm"),
-    plot.margin = unit(c(1, 1, 1, 1), "lines"),
-    legend.text = element_text(size = 16),
-    legend.title = element_blank()#,
-    # legend.position = "none"
-  )
+    ggplot(df, aes(x = group.id, fill = Status)) +
+      geom_rect(
+        # the bars
+        aes(
+          xmin = group.id - 0.25,
+          # control bar gap width
+          xmax = group.id + 0.25,
+          ymin = ystart,
+          ymax = yend
+        ),
+        colour = "#4e4d47",
+        linewidth = .2,
+        alpha = 0.8
+      ) +
+      geom_segment(
+        # the connecting segments between bars
+        data = df %>%
+          filter(group.id == group.id) %>%
+          group_by(group.id) %>%
+          slice_tail() %>%
+          ungroup() %>%
+          select(group.id, yend),
+        aes(
+          x = group.id,
+          xend = if_else(group.id == last(group.id), last(group.id), group.id + 1),
+          y = yend,
+          yend = yend
+        ),
+        linewidth = .3,
+        colour = "gray25",
+        linetype = "dashed",
+        show.legend = FALSE,
+        inherit.aes = FALSE
+      ) +
+      ggrepel::geom_text_repel(
+        # the labels
+        aes(
+          x = group.id,
+          label = paste0(scales::comma(abs(values))),
+          y = rowSums(cbind(ystart, values / 2)),
+          segment.colour = "gray33"
+        ),
+        nudge_x = -.5,
+        arrow = arrow(type = "open", length = unit(.1, "inches")),
+        colour = "#4e4d47",
+        # alpha = .7,
+        size = 5,
+        inherit.aes = FALSE
+      ) +
+      annotate(
+        geom = "text",
+        x = mid_plot,
+        y = max(df$yend) * 1.1,
+        label = paste0(
+          "Total ",
+          getNameByValue(syso_level_of_detail, input$syso_level_of_detail),
```

Comment:
A couple thoughts here:

1. Do you know how I can use `annotate()` in the All Served chart, since the axes aren't continuous, so I can't set the Y value in the same way?
2. When Level of Detail = "All" won't this say "Total All"? That's what happened to me when I tried this method. I think we may need this: 
```
label = paste0(
        "Total ",
        if_else(
          getNameByValue(syso_level_of_detail, input$syso_level_of_detail) == "All",
          "People",
          getNameByValue(syso_level_of_detail, input$syso_level_of_detail)
        ),
        ": ",
        total_clients,
      ),
      size = 16
    ) 
```
I think we should set the size to be larger than the default, too. I used `size=16`

Reply:
1. we talked about earlier :)
2. that would return "Total People" if the input level of detail is "All", otherwise it would return "Total" plus whatever the level of detail is. In the browser I was testing in, these aligned, at least at the time of this writing, maybe some code was merged in that's causing an issue. I'll take another look!
Thanks @alex-silverman!

Reply:
can't recreate this, let me know if you're seeing an issue on your side